extractor/filesystem/sbom/cdx/cdx.go

```diff
@@ -116,40 +116,58 @@ func (e Extractor) convertCdxBomToInventory(cdxBom *cyclonedx.BOM, path string)
 		return results, nil
 	}
 
-	for _, cdxPkg := range *cdxBom.Components {
-		inv := &extractor.Inventory{
-			Locations: []string{path},
-			Metadata:  &Metadata{},
-		}
-		m := inv.Metadata.(*Metadata)
-		inv.Name = cdxPkg.Name
-		inv.Version = cdxPkg.Version
-		if cdxPkg.CPE != "" {
-			m.CPEs = append(m.CPEs, cdxPkg.CPE)
-		}
-		if cdxPkg.PackageURL != "" {
-			packageURL, err := purl.FromString(cdxPkg.PackageURL)
-			if err != nil {
-				log.Warnf("Invalid PURL %q for package ref: %q", cdxPkg.PackageURL, cdxPkg.BOMRef)
-			} else {
-				m.PURL = &packageURL
-				if inv.Name == "" {
-					inv.Name = packageURL.Name
-				}
-				if inv.Version == "" {
-					inv.Version = packageURL.Version
-				}
+	enumerateComponents(*cdxBom.Components, &results)
+
+	for i := range results {
+		results[i].Locations = []string{path}
+	}
+
+	return results, nil
+}
+
+func convertComponentToInventory(cdxPkg cyclonedx.Component) *extractor.Inventory {
+	inv := &extractor.Inventory{
+		Metadata: &Metadata{},
+	}
+	m := inv.Metadata.(*Metadata)
+	inv.Name = cdxPkg.Name
+	inv.Version = cdxPkg.Version
+	if cdxPkg.CPE != "" {
+		m.CPEs = append(m.CPEs, cdxPkg.CPE)
+	}
+	if cdxPkg.PackageURL != "" {
+		packageURL, err := purl.FromString(cdxPkg.PackageURL)
+		if err != nil {
+			log.Warnf("Invalid PURL %q for package ref: %q", cdxPkg.PackageURL, cdxPkg.BOMRef)
+		} else {
+			m.PURL = &packageURL
+			if inv.Name == "" {
+				inv.Name = packageURL.Name
+			}
+			if inv.Version == "" {
+				inv.Version = packageURL.Version
 			}
 		}
-		inv.Metadata = m
-		if m.PURL == nil && len(m.CPEs) == 0 {
-			log.Warnf("Neither CPE nor PURL found for package: %+v", cdxPkg)
-			continue
-		}
-		results = append(results, inv)
+	}
+	inv.Metadata = m
+	if m.PURL == nil && len(m.CPEs) == 0 {
+		log.Warnf("Neither CPE nor PURL found for package: %+v", cdxPkg)
+		return nil
 	}
 
-	return results, nil
+	return inv
+}
+
+func enumerateComponents(components []cyclonedx.Component, results *[]*extractor.Inventory) {
```

Comment:
Move above convertComponentToInventory

From the new style guide (https://github.com/google/osv-scalibr/pull/576/files#diff-caeafed81cc2695ba649974d20ed631f6175fe666a7460335edf18c16b1958d2): "Function ordering: General to specific